### PR TITLE
feat: Add endpoint to the OC challenge service to get challenges per year data

### LINKED ISSUE #2018: Add endpoint to the OC challenge service to get challenges per year data

### PR DESCRIPTION
**Pull request by tschaffter**
- Closes #2018 
- Depends on #2015

## Notes

- The plot data are hardcoded in the challenge service. A future PR will update the challenge service to generate these data dynamically, likely using the Hibernate library that relies on Elasticsearch to perform efficient searches - unless JPA provides a more developer-friendly solution with similar performance.

## TODO

> **Warning**
> The description of the API endpoint should be updated to reflect that we are now reporting the total number of challenges tracked and not the number of challenges that started a given year. Done!

## References

- https://www.baeldung.com/jpa-hibernate-projections#jpaaggregate (JPA)
- https://www.baeldung.com/jpa-hibernate-projections#hibernateaggregate (Hibernate)

## Preview

The request shown below works when sent 1) directly to the challenge service and 2) via the API gateway.

```
GET {{basePath}}/challengeAnalytics/challengesPerYear

HTTP/1.1 200
{
  "years": [
    "1969",
    "2011",
    "2012",
    "2013",
    "2014",
    "2015",
    "2016",
    "2017",
    "2018",
    "2019",
    "2020",
    "2021",
    "2022",
    "2023"
  ],
  "challengeCounts": [
    11,
    12,
    18,
    22,
    28,
    35,
    44,
    69,
    80,
    99,
    117,
    159,
    181,
    199
  ]
}
```

### CODE REVIEW

File: apps/openchallenges/challenge-service/src/main/java/org/sagebionetworks/openchallenges/challenge/service/service/ChallengeAnalyticsService.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class ChallengeAnalyticsService {
+
+  @Transactional(readOnly = true)
+  public ChallengesPerYearDto getChallengesPerYear() {
+    List<String> years =
+        Arrays.asList(
+            "1969", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018", "2019", "2020",
```
Comment (rrchai) [inline apps/openchallenges/challenge-service/src/main/java/org/sagebionetworks/openchallenges/challenge/service/service/ChallengeAnalyticsService.java:16]: Thanks for feeding static challenge counts per year. 
`1969` year is suspicious, so I realize it was a miscalculation caused by undefined dates. I will look into this.

Comment (rrchai) [reply apps/openchallenges/challenge-service/src/main/java/org/sagebionetworks/openchallenges/challenge/service/service/ChallengeAnalyticsService.java:16]: ### Updated data
```json
{
    "years": [
        "2011",
        "2012",
        "2013",
        "2014",
        "2015",
        "2016",
        "2017",
        "2018",
        "2019",
        "2020",
        "2021",
        "2022",
        "2023"
    ],
    "challengeCounts": [
        1,
        6,
        4,
        6,
        7,
        9,
        25,
        11,
        19,
        18,
        42,
        22,
        18
    ]
}
```

**Comment by tschaffter** [comment]
I kept the same endpoint name, just updated its description. I update the API description, the challenge service and the API client for Angular.

File: apps/openchallenges/challenge-service/src/main/java/org/sagebionetworks/openchallenges/challenge/service/service/ChallengeAnalyticsService.java (hunk 1 above)
Comment (tschaffter) [reply apps/openchallenges/challenge-service/src/main/java/org/sagebionetworks/openchallenges/challenge/service/service/ChallengeAnalyticsService.java]: Fixed

**Review by rrchai — APPROVED** [review]
LGTM

**Pull request merged**